src/main/scala/org/hammerlab/guacamole/reads/MDTagUtils.scala

```diff
@@ -75,7 +76,42 @@ object MDTagUtils {
       }
     })
 
-    reference.elems
+    reference.toArray
+  }
+
+  def getReference(read: MappedRead, allowNBase: Boolean): Seq[Byte] = {
+    getReference(read.mdTag, read.sequence, read.cigar, read.start, allowNBase)
+  }
+
+  /**
+   * Rebuilds the reference from a set of overlapping and sorted reads
+   * Fill in N if their is a gap in the reads
```

Comment:
woops... will fix, also not sure why the test are not passing, but doesn't seem like any fail it just dies.  will push a force fix so it should re-test